src/test/test-id128.c

```diff
@@ -145,11 +145,13 @@ TEST(id128) {
         assert_se(id128_read_fd(fd, ID128_UUID, &id2) >= 0);
         assert_se(sd_id128_equal(id, id2));
 
-        assert_se(sd_id128_get_machine_app_specific(SD_ID128_MAKE(f0,3d,aa,eb,1c,33,4b,43,a7,32,17,29,44,bf,77,2e), &id) >= 0);
-        assert_se(sd_id128_get_machine_app_specific(SD_ID128_MAKE(f0,3d,aa,eb,1c,33,4b,43,a7,32,17,29,44,bf,77,2e), &id2) >= 0);
-        assert_se(sd_id128_equal(id, id2));
-        assert_se(sd_id128_get_machine_app_specific(SD_ID128_MAKE(51,df,0b,4b,c3,b0,4c,97,80,e2,99,b9,8c,a3,73,b8), &id2) >= 0);
-        assert_se(!sd_id128_equal(id, id2));
+        if (access("/etc/machine-id", F_OK) >= 0) {
```

Comment:
This doesn't cater to -ENOMEDIUM. We use `if (!sd_booted())` above… I think it'd be nicer to use the same condition.

Reply:
sd_booted just checks for /run/systemd/system, machine-id doesn't necessarily exists if that is true